File: 5.LongestPalindromicSubstring/expand-centers.js
function longestPalindrome(s) {
  let ans = "";

  function expand(left, right) {
    while (left >= 0 && right < s.length && s[left] === s[right]) {
      left--;
      right++;
    }

    return s.slice(left + 1, right);
  }

  for (let i = 0; i < s.length; i++) {
    let odd = expand(i, i);
  }
}

File: 647.PalindromicSubstrings/ExpandCenters.java
class ExpandCenters {
  public int countSubstrings(String s) {
    int n = s.length();
    int count = 0;

    for (int center = 0; center < n; center++) {
      // Odd-length palindromes, center at (center, center)
      count += expandFromCenter(s, center, center);
      // Even-length palindromes, center at (center, center+1)
      count += expandFromCenter(s, center, center + 1);
    }

    return count;
  }

  private int expandFromCenter(String s, int left, int right) {
    int n = s.length();
    int localCount = 0;

    while (left >= 0 && right < n && s.charAt(left) == s.charAt(right)) {
      localCount++;
      left--;
      right++;
    }

    return localCount;
  }
}
